src/deadline_worker_agent/sessions/actions/run_attachment_upload.py

```diff
@@ -0,0 +1,191 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+from __future__ import annotations
+from concurrent.futures import (
+    Executor,
+)
+import os
+import sys
+from logging import LoggerAdapter
+from typing import Any, TYPE_CHECKING, Optional
+from pathlib import Path
+from deadline.job_attachments.models import (
+    JobAttachmentS3Settings,
+)
+from openjd.sessions import LOG as OPENJD_LOG, LogContent
+from openjd.model.v2023_09 import (
+    EmbeddedFileTypes as EmbeddedFileTypes_2023_09,
+    EmbeddedFileText as EmbeddedFileText_2023_09,
+    Action as Action_2023_09,
+    StepScript as StepScript_2023_09,
+    StepActions as StepActions_2023_09,
+)
+from openjd.model import ParameterValue
+from deadline.job_attachments.api.manifest import _manifest_snapshot
```

Comment:
This could cause an import error in windows

```
    Python could not import the service's module 
    Traceback (most recent call last):
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\__init__.py", line 6, in <module>
        from .installer import install
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\installer\__init__.py", line 15, in <module>
        from deadline_worker_agent.installer.win_installer import (
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\installer\win_installer.py", line 36, in <module>
        from ..windows.win_service import WorkerAgentWindowsService
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\windows\win_service.py", line 14, in <module>
        from deadline_worker_agent.startup.entrypoint import entrypoint
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\startup\entrypoint.py", line 24, in <module>
        from ..worker import Worker
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\worker.py", line 25, in <module>
        from .scheduler import WorkerScheduler
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\scheduler\__init__.py", line 4, in <module>
        from .session_queue import SessionActionQueue
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\scheduler\session_queue.py", line 29, in <module>
        from ..sessions.actions import (
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\sessions\actions\__init__.py", line 10, in <module>
        from .run_attachment_upload import AttachmentUploadAction
      File "C:\Program Files\Python311\Lib\site-packages\deadline_worker_agent\sessions\actions\run_attachment_upload.py", line 24, in <module>
        from deadline.job_attachments.api.manifest import _manifest_snapshot
      File "C:\Program Files\Python311\Lib\site-packages\deadline\job_attachments\api\__init__.py", line 5, in <module>
        from .attachment import attachment_download, attachment_upload
      File "C:\Program Files\Python311\Lib\site-packages\deadline\job_attachments\api\attachment.py", line 18, in <module>
        from deadline.client.cli._groups.click_logger import ClickLogger
      File "C:\Program Files\Python311\Lib\site-packages\deadline\client\cli\__init__.py", line 6, in <module>
        from ._deadline_cli import main
      File "C:\Program Files\Python311\Lib\site-packages\deadline\client\cli\_deadline_cli.py", line 14, in <module>
        from ._common import _PROMPT_WHEN_COMPLETE
      File "C:\Program Files\Python311\Lib\site-packages\deadline\client\cli\_common.py", line 31, in <module>
        sigint_handler = SigIntHandler()
                         ^^^^^^^^^^^^^^^
      File "C:\Program Files\Python311\Lib\site-packages\deadline\client\cli\_groups\_sigint_handler.py", line 19, in __new__
        signal.signal(signal.SIGINT, cls._instance._handle_sigint)
      File "C:\Program Files\Python311\Lib\signal.py", line 58, in signal
        handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ValueError: signal only works in main thread of the main interpreter
     
```